livekit-agents/livekit/agents/log.py

```diff
@@ -1,6 +1,27 @@
 import logging
+from typing import Any
 
 DEV_LEVEL = 23
 logging.addLevelName(DEV_LEVEL, "DEV")
 
+
+class FieldsLogger(logging.LoggerAdapter):
+    """A logger adapter that adds fields to each log message"""
+
+    def __init__(self, logger: logging.Logger, fields: dict[str, Any] = {}) -> None:
+        super().__init__(logger, fields)
+
+    def process(self, msg, kwargs):
+        # Merge global fields with any per-message extras
+        extras = self.extra
+        if "extra" in kwargs:
+            extras = self.extra.copy()
+            extras.update(kwargs["extra"])
+        kwargs["extra"] = extras
+        return msg, kwargs
+
+    def with_fields(self, fields: dict[str, Any]) -> "FieldsLogger":
+        return FieldsLogger(self.logger, {**self.extra, **fields})
```

Comment:
I don't think this is needed if we allow people to use their own loggers.
The relevant thing to do would be to create an example that use a custom logger for everything (even plugins)

Reply:
I'm not sure that I agree.. the loggers in python are pretty primitive. this is just porting over the pattern that has worked well for us on the Go side here.

if folks are using their own loggers, there still isn't a simple way to attach fields to it so that they are logged each time.. even our own implementation of adding pid feels hacky.